src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -4,7 +4,7 @@ import {
   ChakraComponent,
   useMultiStyleConfig,
 } from "@chakra-ui/react";
-import React, { useState, forwardRef, useRef } from "react";
+import React, { forwardRef, useRef, useState } from "react";
```

Comment:
That's a good point. If it involves having state or ref we should talk about this because it's getting overly complex and from what I understood, accordions don't generally behave this way. So if we don't have to implement this then perhaps we should not.

Reply:
We don't need it for `Accordion`, but we do need it for `MultiSelect`.

Reply:
That's true but the `Accordion` is a big part of the `MultiSelect` and, to me, `MultiSelect` is inheriting all the features of `Accordion`. There is currently a ref that is being passed from the `MultiSelect` to the `Accordion` to keep track of focus for when the items count button is clicked. This could be used as well when the esc key is pressed. But it we have to keep track of pressing the ESC key in the MultiSelect to close the Accordion, then it feels like the work is being duplicated because the `Accordion` already does check for when the ESC key is pressed.

Perhaps we should reconsider and move all the logic _just_ to the `MultiSelect` from the `Accordion`. @oliviawongnyc  As you try to implement the focus management in the `Accordion`, let me know if my comment makes sense.

Reply:
Okay @EdwinGuzman and @bigfishdesign13, focus should now return to the button when the `Accordion` (or `MultiSelect`) is closed with the `esc` key.